Bottom: 2ee855a0d4f27fcdc2067fbfe1574e838fe8651f
Top:    83c22c229aa74e27d0bebca0fa8b3ec708952ef9
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-01-06 17:22:04 +0000

TEMP: Fixup fallout from nogui change

*** Might be nicer to just implement a no-op end_layer function for each HID_DRAW?


---

diff --git a/src/hid/batch/batch.c b/src/hid/batch/batch.c
index 4f0acb3..2e7bdc2 100644
--- a/src/hid/batch/batch.c
+++ b/src/hid/batch/batch.c
@@ -361,6 +361,7 @@ hid_batch_init ()
 
   batch_hid.graphics              = &batch_graphics;
 
+  common_nogui_graphics_class_init (&batch_graphics_class);
   common_draw_helpers_class_init (&batch_graphics_class);
 
   batch_graphics_class.set_layer      = batch_set_layer;
@@ -379,6 +380,7 @@ hid_batch_init ()
   batch_graphics_class.fill_rect      = batch_fill_rect;
 
   batch_graphics.klass = &batch_graphics_class;
+  common_nogui_graphics_init (&batch_graphics);
   common_draw_helpers_init (&batch_graphics);
 
   hid_register_hid (&batch_hid);
diff --git a/src/hid/bom/bom.c b/src/hid/bom/bom.c
index 9c81682..d17aea7 100644
--- a/src/hid/bom/bom.c
+++ b/src/hid/bom/bom.c
@@ -15,6 +15,7 @@
 #include "pcb-printf.h"
 
 #include "hid.h"
+#include "hid_draw.h"
 #include "hid/common/hidnogui.h"
 #include "../hidint.h"
 
diff --git a/src/hid/common/hidinit.c b/src/hid/common/hidinit.c
index 9f05800..b964d24 100644
--- a/src/hid/common/hidinit.c
+++ b/src/hid/common/hidinit.c
@@ -21,6 +21,7 @@
 
 #include "global.h"
 #include "hid.h"
+#include "hid_draw.h"
 #include "hidnogui.h"
 #include "../hidint.h"
 
diff --git a/src/hid/common/hidnogui.c b/src/hid/common/hidnogui.c
index 85e678d..2618d98 100644
--- a/src/hid/common/hidnogui.c
+++ b/src/hid/common/hidnogui.c
@@ -476,7 +476,7 @@ common_nogui_init (HID *hid)
   hid->finish_debug_draw =    nogui_finish_debug_draw;
 }
 
-static void
+void
 common_nogui_graphics_class_init (HID_DRAW_CLASS *klass)
 {
   klass->set_layer =       nogui_set_layer;
@@ -503,7 +503,7 @@ common_nogui_graphics_class_init (HID_DRAW_CLASS *klass)
   klass->thindraw_pcb_pv =  nogui_thindraw_pcb_pv;
 }
 
-static void
+void
 common_nogui_graphics_init (HID_DRAW *graphics)
 {
 }
diff --git a/src/hid/common/hidnogui.h b/src/hid/common/hidnogui.h
index 518ff06..e8acba0 100644
--- a/src/hid/common/hidnogui.h
+++ b/src/hid/common/hidnogui.h
@@ -2,6 +2,8 @@
 #define PCB_HID_COMMON_HIDNOGUI_H
 
 void common_nogui_init (HID *hid);
+void common_nogui_graphics_class_init (HID_DRAW_CLASS *klass);
+void common_nogui_graphics_init (HID_DRAW *graphics);
 HID *hid_nogui_get_hid (void);
 
 #endif
diff --git a/src/hid/gcode/gcode.c b/src/hid/gcode/gcode.c
index a68e86e..3a7c0c2 100644
--- a/src/hid/gcode/gcode.c
+++ b/src/hid/gcode/gcode.c
@@ -1622,6 +1622,7 @@ hid_gcode_init ()
 
   gcode_hid.graphics            = &gcode_graphics;
 
+  common_nogui_graphics_class_init (&gcode_graphics_class);
   common_draw_helpers_class_init (&gcode_graphics_class);
 
   gcode_graphics_class.set_layer      = gcode_set_layer;
@@ -1642,6 +1643,7 @@ hid_gcode_init ()
 
   gcode_graphics.klass = &gcode_graphics_class;
   gcode_graphics.poly_before = true;
+  common_nogui_graphics_init (&gcode_graphics);
   common_draw_helpers_init (&gcode_graphics);
 
   hid_register_hid (&gcode_hid);
diff --git a/src/hid/gerber/gerber.c b/src/hid/gerber/gerber.c
index 22ae4d6..88393ef 100644
--- a/src/hid/gerber/gerber.c
+++ b/src/hid/gerber/gerber.c
@@ -1360,6 +1360,7 @@ hid_gerber_init ()
 
   gerber_hid.graphics            = &gerber_graphics;
 
+  common_nogui_graphics_class_init (&gerber_graphics_class);
   common_draw_helpers_class_init (&gerber_graphics_class);
 
   gerber_graphics_class.set_layer      = gerber_set_layer;
@@ -1378,6 +1379,7 @@ hid_gerber_init ()
   gerber_graphics_class.fill_rect      = gerber_fill_rect;
 
   gerber_graphics.klass = &gerber_graphics_class;
+  common_nogui_graphics_init (&gerber_graphics);
   common_draw_helpers_init (&gerber_graphics);
 
   hid_register_hid (&gerber_hid);
diff --git a/src/hid/gtk/gtkhid-main.c b/src/hid/gtk/gtkhid-main.c
index 20becc7..190a1cd 100644
--- a/src/hid/gtk/gtkhid-main.c
+++ b/src/hid/gtk/gtkhid-main.c
@@ -2182,6 +2182,7 @@ hid_gtk_init ()
 
   ghid_hid.graphics                 = &ghid_graphics;
 
+  common_nogui_graphics_class_init (&ghid_graphics_class);
   common_draw_helpers_class_init (&ghid_graphics_class);
 
   ghid_graphics_class.set_layer      = ghid_set_layer;
@@ -2205,6 +2206,7 @@ hid_gtk_init ()
 
   ghid_graphics.klass = &ghid_graphics_class;
   ghid_graphics.poly_after = true;
+  common_nogui_graphics_init (&ghid_graphics);
   common_draw_helpers_init (&ghid_graphics);
 
   hid_register_hid (&ghid_hid);
diff --git a/src/hid/lesstif/main.c b/src/hid/lesstif/main.c
index 889064a..f5c21bb 100644
--- a/src/hid/lesstif/main.c
+++ b/src/hid/lesstif/main.c
@@ -4113,6 +4113,7 @@ hid_lesstif_init ()
 
   lesstif_hid.graphics                = &lesstif_graphics;
 
+  common_nogui_graphics_class_init (&lesstif_graphics_class);
   common_draw_helpers_class_init (&lesstif_graphics_class);
 
   lesstif_graphics_class.set_layer      = lesstif_set_layer;
@@ -4134,6 +4135,7 @@ hid_lesstif_init ()
 
   lesstif_graphics.klass = &lesstif_graphics_class;
   lesstif_graphics.poly_before = true;
+  common_nogui_graphics_init (&lesstif_graphics);
   common_draw_helpers_init (&lesstif_graphics);
 
   hid_register_hid (&lesstif_hid);
diff --git a/src/hid/nelma/nelma.c b/src/hid/nelma/nelma.c
index dff74b0..8940a70 100644
--- a/src/hid/nelma/nelma.c
+++ b/src/hid/nelma/nelma.c
@@ -1079,6 +1079,7 @@ hid_nelma_init()
 
   nelma_hid.graphics            = &nelma_graphics;
 
+  common_nogui_graphics_class_init (&nelma_graphics_class);
   common_draw_helpers_class_init (&nelma_graphics_class);
 
   nelma_graphics_class.set_layer      = nelma_set_layer;
@@ -1099,6 +1100,7 @@ hid_nelma_init()
 
   nelma_graphics.klass = &nelma_graphics_class;
   nelma_graphics.poly_before = true;
+  common_nogui_graphics_init (&nelma_graphics);
   common_draw_helpers_init (&nelma_graphics);
 
   hid_register_hid (&nelma_hid);
diff --git a/src/hid/png/png.c b/src/hid/png/png.c
index 00dd54c..1260d70 100644
--- a/src/hid/png/png.c
+++ b/src/hid/png/png.c
@@ -1854,6 +1854,7 @@ hid_png_init ()
 
   png_hid.graphics            = &png_graphics;
 
+  common_nogui_graphics_class_init (&png_graphics_class);
   common_draw_helpers_class_init (&png_graphics_class);
 
   png_graphics_class.set_layer      = png_set_layer;
@@ -1873,6 +1874,7 @@ hid_png_init ()
 
   png_graphics.klass = &png_graphics_class;
   png_graphics.poly_before = true;
+  common_nogui_graphics_init (&png_graphics);
   common_draw_helpers_init (&png_graphics);
 
 #ifdef HAVE_SOME_FORMAT
diff --git a/src/hid/ps/eps.c b/src/hid/ps/eps.c
index 3f0afc6..5bf65c8 100644
--- a/src/hid/ps/eps.c
+++ b/src/hid/ps/eps.c
@@ -695,6 +695,7 @@ hid_eps_init ()
 
   eps_hid.graphics            = &eps_graphics;
 
+  common_nogui_graphics_class_init (&eps_graphics_class);
   common_draw_helpers_class_init (&eps_graphics_class);
 
   eps_graphics_class.set_layer      = eps_set_layer;
@@ -714,6 +715,7 @@ hid_eps_init ()
 
   eps_graphics.klass = &eps_graphics_class;
   eps_graphics.poly_after = true;
+  common_nogui_graphics_init (&eps_graphics);
   common_draw_helpers_init (&eps_graphics);
 
   hid_register_hid (&eps_hid);
diff --git a/src/hid/ps/ps.c b/src/hid/ps/ps.c
index 1572ccc..f8df547 100644
--- a/src/hid/ps/ps.c
+++ b/src/hid/ps/ps.c
@@ -1573,11 +1573,13 @@ hid_ps_init ()
 
   ps_hid.graphics           = &ps_graphics;
 
+  common_nogui_graphics_class_init (&ps_graphics_class);
   common_draw_helpers_class_init (&ps_graphics_class);
   ps_ps_graphics_class_init (&ps_graphics_class);
 
   ps_graphics.klass = &ps_graphics_class;
   ps_graphics.poly_before = true;
+  common_nogui_graphics_init (&ps_graphics);
   common_draw_helpers_init (&ps_graphics);
   ps_ps_graphics_init (&ps_graphics);
